src/include/rados/librados.hpp

```diff
@@ -28,6 +28,9 @@ namespace librados
   class ObjectOperationImpl;
   struct ObjListCtx;
   struct PoolAsyncCompletionImpl;
+  class CacheRequest;
```

Comment:
This stuff shouldn't be included in librados.hpp itself. It should be split out into its own library that uses librados, similarly to what libradosstriper does.